server/sonar-server/src/main/java/org/sonar/server/issue/IssueQueryFactory.java

```diff
@@ -200,35 +190,38 @@ private Date findCreatedAfterFromComponentUuid(DbSession dbSession, String uuid)
     return assignees;
   }
 
-  private boolean mergeDeprecatedComponentParameters(DbSession session, @Nullable Boolean onComponentOnly,
-    @Nullable Collection<String> components,
-    @Nullable Collection<String> componentUuids,
-    @Nullable Collection<String> componentKeys,
-    @Nullable Collection<String> componentRootUuids,
-    @Nullable Collection<String> componentRoots,
-    Set<String> allComponentUuids) {
+  private boolean mergeDeprecatedComponentParameters(DbSession session, SearchWsRequest request, List<ComponentDto> allComponents) {
+    Boolean onComponentOnly = request.getOnComponentOnly();
+    Collection<String> components = request.getComponents();
+    Collection<String> componentUuids = request.getComponentUuids();
+    Collection<String> componentKeys = request.getComponentKeys();
+    Collection<String> componentRootUuids = request.getComponentRootUuids();
+    Collection<String> componentRoots = request.getComponentRoots();
+    String branch = request.getBranch();
+
     boolean effectiveOnComponentOnly = false;
 
     checkArgument(atMostOneNonNullElement(components, componentUuids, componentKeys, componentRootUuids, componentRoots),
       "At most one of the following parameters can be provided: %s, %s, %s, %s, %s",
       PARAM_COMPONENT_KEYS, PARAM_COMPONENT_UUIDS, PARAM_COMPONENTS, PARAM_COMPONENT_ROOTS, PARAM_COMPONENT_UUIDS);
 
     if (componentRootUuids != null) {
-      allComponentUuids.addAll(componentRootUuids);
+      allComponents.addAll(getComponentsFromUuids(session, componentRootUuids));
       effectiveOnComponentOnly = false;
     } else if (componentRoots != null) {
-      allComponentUuids.addAll(convertComponentKeysToUuids(session, componentRoots));
+      allComponents.addAll(getComponentsFromKeys(session, componentRoots, branch));
       effectiveOnComponentOnly = false;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment; "effectiveOnComponentOnly" already holds the assigned value along all execution paths. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS4165)